### PR TITLE
Take 2: Qt6 readiness: Use typed enums

### PR DESCRIPTION
**Pull request by jonoomph**
Updating code base (especially Qt enums) to be compatible with Qt6. This is mostly work from @ferdnyc and PR https://github.com/OpenShot/openshot-qt/pull/4833. I've made a few additional changes (formatting, bug fixes), but due to the older base of this PR (I think), I had trouble merging these changes back into the original PR.

Additional changes:
- Removed some unused lines of code
- Removed some unused imports
- Reformatted a few small blocks of code
- Refactored a few small blocks
- Found some duplicate code and removed it

I also did a bunch of local testing with this branch, using Qt5, and did not see any breakages, errors, or stack traces.

### CODE REVIEW

**Comment by jonoomph** [comment]
[PR: Support for PyQt6 (initial steps)](https://trello.com/c/qhKzSLUJ/149-pr-support-for-pyqt6-initial-steps)

**Comment by ferdnyc** [comment]
@jonoomph 

Honestly? This doesn't sit right with me at all.

Never mind the pointlessness of leaving review comments on my other PR, then closing it before I could even address them. That's actually beside the point.

If you wanted to make changes to my commit, there are two different ways — no, wait, _three_ different ways — you could do that without obliterating the attribution. (Proper git attribution, not a note in a commit message.)

1. You could've created this branch, edited my PR to target it, and merged the PR to this branch
2. You could've `git cherry-pick`ed the commit from the PR branch in my fork into this branch, after creating it.
3. You could've used `git format-patch -1` from a checkout of my PR branch, to export my commit to a patch file, then run `git am 0001-Py-Qt6-readiness-Use-typed-enums.patch` from this branch to re-import the commit.

Any of those would keep my ID attached to my changes.

File: src/windows/export.py [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ def titlestring(sec, fps, mess):
             # Make progress bar green (to indicate we are done)
             from PyQt5.QtGui import QPalette
             p = QPalette()
-            p.setColor(QPalette.Highlight, Qt.green)
+            p.setColor(QPalette.Highlight, Qt.GlobalColor.green)
```
Comment (JacksonRG) [inline src/windows/export.py:1012]: I saw other `QPalette.Highlight` replaced with `QPalette.ColorRole.Highlight`

**Review by JacksonRG — CHANGES_REQUESTED** [review]
One line looks like it's missing an enum class. Other than that, looks good.

**Comment by jonoomph** [comment]
> @jonoomph
> 
> Honestly? This doesn't sit right with me at all.
> 
> Never mind the pointlessness of leaving review comments on my other PR, then closing it before I could even address them. That's actually beside the point.
> 
> If you wanted to make changes to my commit, there are two different ways — no, wait, _three_ different ways — you could do that without obliterating the attribution. (Proper git attribution, not a note in a commit message.)
> 
> 1. You could've created this branch, edited my PR to target it, and merged the PR to this branch
> 2. You could've `git cherry-pick`ed the commit from the PR branch in my fork into this branch, after creating it.
> 3. You could've used `git format-patch -1` from a checkout of my PR branch, to export my commit to a patch file, then run `git am 0001-Py-Qt6-readiness-Use-typed-enums.patch` from this branch to re-import the commit.
> 
> Any of those would keep my ID attached to my changes.

@ferdnyc Please calm down, no need to get aggressive. Nobody is trying to steal your attribution. I am fine merging your previous PR, if you can pull my changes from this branch and fix all the issues with the original PR. I spent an hour trying to merge changes into your branch unsuccessfully (not to mention prioritized this work above my own), and thus, created this one as a last ditch effort to move this forward. So, you can criticize my lack of git skills, sure, but let's not be overly emotional and accusatory. 

Closing this one for now (waiting on changes to the original)

**Pull request closed without merging**